torch/ao/quantization/fx/_model_report/_detector.py

```diff
@@ -134,9 +117,147 @@ def _detect_per_channel_helper(model: nn.Module):
+        Otherwise recommend static quantization
+
+    This will then generate suggestions for dynamic vs static quantization focused around Linear.
+
+    Args:
+        model (GraphModule): The prepared and calibrated GraphModule with inserted ModelReportObservers around layers of interest
+        tolerance (float, optional): The threshold where S metric is stationary above and non-stationary otherwise. Default: 0.5
+
+    Returns a tuple with two elements:
+        String report of of whether dynamic or static quantization is recommended for certain modules
+        Dictionary mapping modules with ModelReportObservers around them to:
+            whether dynamic quantization is recommended
+            their S metric of input to module
+            whether input to module is stationary or non-stationary
+            their S metric of output of module
+            whether output of module is stationary or non-stationary
+            the tolerance level to decided whether input/output is stationary or non-stationary
+    """
+
+    # store modules dynamic vs static information
+    module_dynamic_static_info = {}
+
+    # loop through all submodules included nested ones
+    for name, module in model.named_modules():
+        # if module has the ModelReportObserver attached to it
+        if hasattr(module, DEFAULT_PRE_OBSERVER_NAME) and hasattr(module, DEFAULT_POST_OBSERVER_NAME):
+            # get pre and post observers for the module
+            pre_obs = getattr(module, DEFAULT_PRE_OBSERVER_NAME)
+            post_obs = getattr(module, DEFAULT_POST_OBSERVER_NAME)
+
+            # get the statistics for each module
+            pre_stat = pre_obs.get_batch_to_epoch_ratio()
+            post_stat = post_obs.get_batch_to_epoch_ratio()
+
+            # record module, pre and post stat, and whether to do dynamic or static based off it
+            # true if post observer data distribution is non-stationary, false if it's stationary
+            dynamic_recommended = post_stat <= tolerance
+
+            # specify the classifications for whether data distributions considered stationary or non-stationary
+            pre_obs_dist_classification = DEFAULT_STATIONARY if pre_stat > tolerance else DEFAULT_NON_STATIONARY
+            post_obs_dist_classification = DEFAULT_STATIONARY if post_stat > tolerance else DEFAULT_NON_STATIONARY
+
+            # store the set of important information for this module
+            module_info = {
+                "tolerance": tolerance,
+                "dynamic_recommended": dynamic_recommended,
+                "pre_observer_comp_stat": pre_stat,
+                "pre_observer_data_dist": pre_obs_dist_classification,
+                "post_observer_comp_stat": post_stat,
+                "post_observer_data_dist": post_obs_dist_classification,
+            }
+
+            module_dynamic_static_info[name] = module_info
```

Comment:
name is not a unique identifier, need to use fqn

Reply:
this actually was the fqn (just printed and verified, renamed var to properly reflect)